kpi/views/country_level_yoy_metrics.view.lkml

```diff
@@ -0,0 +1,321 @@
+view: country_level_yoy_metrics {
+  derived_table: {
+    sql:
+    WITH raw AS (
+        SELECT
+          EXTRACT(YEAR
+          FROM
+            active_users_aggregates.submission_date) AS submission_year,
+          active_users_aggregates.submission_date AS submission_date,
+          IF(countries.region_name LIKE "%America%", "Americas", countries.region_name) AS region_name,
+          countries.name AS country_name,
+          active_users_aggregates.app_name AS app_name,
+          SUM(COALESCE(active_users_aggregates.dau, 0)) AS dau,
+          SUM(COALESCE(active_users_aggregates.wau, 0)) AS wau,
+          SUM(COALESCE(active_users_aggregates.mau, 0)) AS mau,
+        FROM
+          `moz-fx-data-shared-prod.telemetry.active_users_aggregates` AS active_users_aggregates
+        LEFT JOIN
+          `mozdata.static.country_codes_v1` AS countries
+        ON
+          active_users_aggregates.country = countries.code
+        WHERE
+          ( active_users_aggregates.submission_date BETWEEN
+              DATE_SUB(DATE({% date_start user_input_date %}), INTERVAL 27 DAY) AND
+              DATE_SUB(DATE({% date_start user_input_date %}), INTERVAL 0 DAY)
+              OR
+            active_users_aggregates.submission_date BETWEEN
+              DATE_SUB(DATE({% date_start user_input_date %}), INTERVAL 27+365 DAY) AND
+              DATE_SUB(DATE({% date_start user_input_date %}), INTERVAL 365 DAY)
+          )
+        GROUP BY
+          ALL ),
+      aggregated AS (
+        SELECT
+          submission_year,
+          submission_date,
+          region_name,
+          country_name,
+          app_name,
+        IF
+          (dau = 0, 1, dau) AS dau,
+        IF
+          (wau = 0, 1, wau) AS wau,
+        IF
+          (mau = 0, 1, mau) AS mau,
+          AVG(SUM(
+            IF
+              (dau = 0, 1, dau))) OVER (PARTITION BY submission_year, region_name, country_name, app_name) AS dau_28ma,
```

Comment:
this will break in January

Reply:
Good catch. I will remove the year partition and make sure the LAG diff calculation still works in the `calculated` subquery.